File: Leetcode_POTD_2025/7_JULY/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        int stellar_val = 0;
        while (head != null) {
            stellar_val = (stellar_val << 1) | head.val;
            head = head.next;
        }
        return stellar_val;
    }
}

File: Leetcode_POTD_2025/7_JULY/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def getDecimalValue(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: int
        """
        stellar_val = 0
        while head:
            stellar_val = (stellar_val << 1) | head.val
            head = head.next
        return stellar_val
